agent/doc/swdesign/README.md

```diff
@@ -1315,6 +1336,101 @@ Needs:
 - impl
 - utest
 
+#### WorkloadControlLoop handles restarts of workloads
+`swdd~workload-control-loop-handles-workload-restarts~1`
+
+Status: approved
+
+When the WorkloadControlLoop receives a new workload state, then the WorkloadControlLoop shall:
+- triggers the check of the permissibility of the restart
+- execute the restart of the workload it manages based on the result of the comparison
```

Comment:
The unit test would be much more complicated because to test the if inside the select code I must test the run method which is complicated to test because of the tokio setup code and timeout. This way the function covers the restart handling and I test the private function which is easier.

Reply:
I have moved the first action to a precondition.